File: sort/Java/InsertSort.java
/**
 * ‘}“üƒ\[ƒg‚ğ‚·
 * @author masao
 *
 */
public class InsertSort {
	
	public static void main(String[] args) {
		
		int[] array = createRandomArray(Integer.parseInt(args[0]));
		
		System.out.println("®—ñ‘O");
		for (int i = 0; i < array.length; i++) {
			System.out.print(array[i] + " ");
		}
		
		insertSort(array);
	}

	private static void insertSort(int[] array) {
		//TODO ƒAƒ‹ƒSƒŠƒYƒ€À‘•
	}

	/**
	 * intŒ^‚Ì”z—ñ‚É—”‚ğŠi”[‚·‚é
	 * @param num
	 * @return
	 */
	private static int[] createRandomArray(int num) {
		
		int[] result = new int[num];
		
		for (int i = 0; i < result.length; i++) {
			result[i] = (int) (Math.random()*10);
		}
		
		return result;
	}

}

File: sort/Ruby/InsertSort.rb
# 挿入ソート(Ruby版)
class InsertSort
  def createRandomArray(num)
    result = []
    for i in 1..num do
      result.push(rand(100))
    end
    return result
  end

  def insertSort(array)
    #挿入ソートの処理を書く
    i = 0
    while i < array.length do
      insert(array, i, array[i])
      i += 1
    end
    return array
  end

  def insert(array, pos, value)
    #挿入ソートの各回の処理
    index = pos -1;
   while index >= 0 && array[index] > value do
      array[index + 1] = array[index];
      index -= 1
    end
    array[index + 1] = value
  end
end

if __FILE__ == $0
  sort = InsertSort.new()
  array = sort.createRandomArray(10)
  p "整列前："
  p array
  sortedArray = sort.insertSort(array)
  p "整列後："
  p sortedArray
end
